tests/imageupload/imageuploadediting.js

```diff
@@ -822,18 +827,27 @@ describe( 'ImageUploadEditing', () => {
 } );
 
 // Asserts actual and expected model data.
-// Note: Since this function is run inside a promise, all errors needs to be caught
-// and rethrow to be correctly processed by a testing framework.
 //
 // @param {function} done Callback function to be called when assertion is done.
 // @param {String} actual Actual model data.
 // @param {String} expected Expected model data.
 function expectModel( done, actual, expected ) {
-	try {
+	tryExpect( done, () => {
 		expect( actual ).to.equal( expected );
-		done();
+	} );
+}
+
+// Runs given expect function in a try-catch. It should be used only when `expect` is called as a result of a `Promise`
+// resolution as all errors may be caught by tested code and needs to be rethrow to be correctly processed by a testing framework.
+//
+// @param {Function} doneFn Function to run when assertion is done.
+// @param {Function} expectFn Function containing all assertions.
+function tryExpect( doneFn, expectFn ) {
```

Comment:
The same thing with `tryExpect` as in ckeditor5-upload PR review. I'd remove it and use promises API from mocha.

Reply:
🤔 Here it is a little more tricky. I have corrected one test to return the promise directly - https://github.com/ckeditor/ckeditor5-image/pull/258/commits/fa309d5c41aa9bf20df20e72352759aefcf96833.

In other cases I don't see a straightforward way to do it, because `tryExpect` is used inside event listeners which are triggered by the promise resolution, however the promise itself is not created in the test, see for example - https://github.com/ckeditor/ckeditor5-image/blob/t/ckeditor5-upload/87/tests/imageupload/imageuploadediting.js#L335-L375.

Reply:
Sure, we can live with that :)